File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/ActionContextFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import hudson.EnvVars;
import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.model.ActionContext;

import java.io.Serializable;

@Extension
public class ActionContextFactory extends InvisibleAction implements Serializable {

    public ActionContext create(EnvVars env) {
        return new ActionContext(
            Boolean.getBoolean(env.get("FORCE_RELEASE", "false"))
        );
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/ActionContextFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.model.ActionContext
import org.gradle.api.provider.Provider
import org.gradle.api.provider.ProviderFactory

class ActionContextFactory {

    fun create(providerFactory: ProviderFactory): ActionContext =
        ActionContext(
            isForceRelease = providerFactory.systemProperty("forceRelease").asBoolean()
        )

    private fun Provider<String>.asBoolean(): Boolean =
        try {
            getOrElse("false")
                .takeIf { it.isNotBlank() }
                ?.toBoolean()
                ?: true
        } catch (_: Exception) {
            false
        }

}
